From a06dd011007f4ad580dfb110945467f69df15fd8 Mon Sep 17 00:00:00 2001
From: Bojan Zivanovic
Date: Sun, 4 Mar 2018 23:19:19 +0100
Subject: Issue #2923337 followup: Fix regression that broke theme suggestions
 for variation field templates.


diff --git a/modules/product/commerce_product.module b/modules/product/commerce_product.module
index 2f6ad46..a94eb1a 100644
--- a/modules/product/commerce_product.module
+++ b/modules/product/commerce_product.module
@@ -44,6 +44,30 @@ function commerce_product_entity_view_display_update(EntityInterface $entity) {
 }
 
 /**
+ * Implements hook_theme_registry_alter().
+ */
+function commerce_product_theme_registry_alter(&$theme_registry) {
+  // The preprocess function must run after quickedit_preprocess_field().
+  $theme_registry['field']['preprocess functions'][] = 'commerce_product_remove_quickedit';
+}
+
+/**
+ * Turn off Quick Edit for injected variation fields, to avoid warnings.
+ */
+function commerce_product_remove_quickedit(&$variables) {
+  $entity_type_id = $variables['element']['#entity_type'];
+  if ($entity_type_id != 'commerce_product_variation' || empty($variables['#ajax_replace_class'])) {
+    return;
+  }
+
+  if (isset($variables['attributes']['data-quickedit-field-id'])) {
+    unset($variables['attributes']['data-quickedit-field-id']);
+    $context_key = array_search('user.permissions', $variables['#cache']['contexts']);
+    unset($variables['#cache']['contexts'][$context_key]);
+  }
+}
+
+/**
  * Implements hook_theme().
  */
 function commerce_product_theme() {
diff --git a/modules/product/src/ProductViewBuilder.php b/modules/product/src/ProductViewBuilder.php
index a78283d..92cb7dc 100644
--- a/modules/product/src/ProductViewBuilder.php
+++ b/modules/product/src/ProductViewBuilder.php
@@ -78,8 +78,6 @@ class ProductViewBuilder extends EntityViewBuilder {
       $attribute_field_names = $variation->getAttributeFieldNames();
       $rendered_fields = $this->variationFieldRenderer->renderFields($variation, $view_mode);
       foreach ($rendered_fields as $field_name => $rendered_field) {
-        // Turn off Quick Edit for injected variation fields, to avoid warnings.
-        $rendered_field['#view_mode'] = '_custom';
         // Group attribute fields to allow them to be excluded together.
         if (in_array($field_name, $attribute_field_names)) {
           $build['variation_attributes']['variation_' . $field_name] = $rendered_field;
-- 
cgit v0.10.2

